lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Azure.DataLake.Store;
+using Microsoft.Azure.DataLake.Store.FileTransfer;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureDataLake
+{
+    /// <summary>
+    /// An IFileSystem implementation for Azure Data Lake Store.
+    /// </summary>
+    internal sealed class AzureDataLakeFileSystem : IFileSystem
+    {
+        private readonly IDataLakeStoreClient _client;
+        private readonly AdlsClient _adlsClient;
+
+        [Inject]
+        private AzureDataLakeFileSystem(IDataLakeStoreClient client)
+        {
+            _client = client;
+            _adlsClient = _client.GetAdlsClient();
+        }
+
+        /// <summary>
+        /// Opens the given URI for reading
+        /// </summary>
+        /// <exception cref="AdlsException">If the URI couldn't be opened.</exception>
+        public Stream Open(Uri fileUri)
+        {
+            return _adlsClient.GetReadStream(fileUri.AbsolutePath);
```

Comment:
AbsolutePath does not contain the prefix of the file system, like "adl://xxxxx/. Where is that information set into this DataLakeFileSystem? AbsoluteUri contains the full path.

Reply:
AdlsClient handles that by itself. I have tested it with real azure data lake and gave a demo to the team showing that this works end to end